### ISSUE TITLE
Placeholder for  `root reference form` in mono-branch cognate sets

### ISSUE THREAD

**Issue by PaulHeggarty** [lingdb/CoBL#364]
- [x] In the `1-Meaning, All Cognate Sets` form (and related forms based on the same info), for any cognate set found in just one branch, and for which the `root reference form` field has not been filled in manually, then it should be filled in by a placeholder calculated automatically as follows.
The placeholder should be a copy of the 'orthographic' form in another language in that branch, with the best candidate languages being ranked in this order:  
1) The reference language for that branch (if it has this cognate set, of course).
2) If this cognate set is not found in (1), then take the first of the preferred 'example' languages in that branch (see issue).
3) If this cognate set is not found in (2), then just the first language within that branch that does have this cognate set.

**Comment by runjak** [comment]
Current placeholder logic for the `Root reference form` entries is implemented in [CognateClass.rootFormOrPlaceholder](https://github.com/lingdb/CoBL/blob/master/ielex/lexicon/models.py#L1238-L1251) like this:
```python
def rootFormOrPlaceholder(self):
    # Added for #246
    # If we have a root_form, we see if we can enrich it:
    if self.root_form != '':
        return self.root_form
    # No root_form or loanword, maybe only one lexeme in cognate class:
    if self.lexeme_set.count() == 1:
        return self.lexeme_set.get().source_form
    # Do we have a loanword?
    if self.loanword:
        if self.sourceFormInLoanLanguage != '':
            return "(%s)" % self.sourceFormInLoanLanguage
    # Nothing left to try:
    return ''
```
It follows the logic discussed in #246 originally.
Currently this placeholder code produces cases like this:
![screenshot_2016-12-06_11-13-55](https://cloud.githubusercontent.com/assets/5417642/20921808/13ce65c8-bba6-11e6-8dfb-db3245224f88.png)
But there are still cases where this computation doesn't turn up with anything:
![screenshot_2016-12-06_11-21-42](https://cloud.githubusercontent.com/assets/5417642/20921849/402a9c18-bba6-11e6-9c35-413f2f46f387.png)
I will extend the placeholder computation to do the additional lookups specified in this issue in case they don't find something with the current logic.

**Comment by runjak** [comment]
I've implemented the requested lookup in commit fc04481ec8bffedefeddfc02b77d85ce395e2d2c.
This makes it possible to add placeholders in `2/3` of the cases missing in the above screenshot:
![screenshot_2016-12-06_12-19-34](https://cloud.githubusercontent.com/assets/5417642/20923831/a024ff84-bbae-11e6-9599-ee4d005e027c.png)
@PaulHeggarty the only open question remaining for me here would be if this lookup order is what we want or whether we want to replace the original placeholder lookup instead.

**Comment by runjak** [comment]
My implementation for this has now been deployed.
But as described in comments above there was a bit of uncertainty
on how to go ahead with the implementation.
So it may make sense to review this.
Therefore I'm mentioning @cormacanderson in addition to Paul.
